src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -483,7 +483,10 @@ uct_dc_mlx5_iface_dci_detach(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep,
 
     ep->dci                      = UCT_DC_MLX5_EP_NO_DCI;
     ep->flags                   &= ~UCT_DC_MLX5_EP_FLAG_TX_WAIT;
-    iface->tx.dcis[dci_index].ep = new_ep;
```

Comment:
can't this create some inconsistency that ep ep no longer points to dci, but dci still points to ep?
or maybe add new function schedule_release which will just UCS_BITMAP_SET and add progress callback to release. and when we want to release the dci from purge context of pending context, just call schedule_release

Reply:
ok, refactored

Reply:
also added dci->ep=NULL